table/block_based_table_builder.cc

```diff
@@ -610,6 +610,16 @@ void BlockBasedTableBuilder::WriteRawBlock(const Slice& block_contents,
         EncodeFixed32(trailer_without_type, XXH32_digest(xxh));
         break;
       }
+      case kxxHash64: {
+        XXH64_state_t* const state = XXH64_createState();
+        XXH64_reset(state, 0);
+        XXH64_update(state, block_contents.data(),
+                static_cast<uint32_t>(block_contents.size()));
+        XXH64_update(state, trailer, 1);  // Extend  to cover block type
+        EncodeFixed32(trailer_without_type, static_cast<uint32_t>(XXH64_digest(state)));
```

Comment:
Just for code readable, I suggest we do an explicit instruction to take the lower bits before casting to uint32_t. Something like `& 0xffffffff`. The result should be the same though.

Reply:
I guess this is a question for @Cyan4973 . Does small endian and big endian machines get the same xxhash64 results? If they are, then 0xffffffff should return the same result.

Reply:
@siying , no matter big endian or small endian, the static_cast<uint32_t> should be ok

Reply:
@Cyan4973 thanks
@jsjhoubo I know that, but if `& 0xffffffff` generate the same result, I suggest we do that before casting to uint32_t. I personally don't like the idea of down casting with the value itself changed. I feel it hurts readability.